File: ts/src/provider/openai-provider.ts
import type { Message, ToolDefinition } from "../schema/message";
import type { LlmProvider } from "./llm-provider";

/** 对照 go/internal/provider/openai.go */
export class OpenaiProvider implements LlmProvider {
  constructor(private readonly model: string) {
    void this.model;
  }

  generate(
    _messages: Message[],
    _availableTools: ToolDefinition[] | null,
  ): Promise<Message> {
    throw new Error("TODO: 对照 go/internal/provider/openai.go 实现");
  }
}

File: python/src/tiny_claw/provider/openai_provider.py
from tiny_claw.provider.base import LlmProvider
from tiny_claw.schema import Message, ToolDefinition


class OpenaiProvider(LlmProvider):
    """对照 go/internal/provider/openai.go"""

    def __init__(self, model: str) -> None:
        self.model = model

    def generate(
        self,
        messages: list[Message],
        available_tools: list[ToolDefinition] | None,
    ) -> Message:
        raise NotImplementedError("TODO: 对照 go/internal/provider/openai.go 实现")
